Add first draft of release notes (#268)
anbox-cloud/discourse-docs: release-notes/1.18.2.md

Please see [component versions](https://anbox-cloud.io/docs/component-versions) for a list of updated components.

## New features & improvements
* Android security updates for July 2023. See [Android Security Bulletin - July 2023](https://source.android.com/docs/security/bulletin/2023-07-01) for more information.
* The Android WebView has been updated to [tba](tba).

## Bug fixes
* When upgrading Anbox Cloud, the `Reboot required to activate new kernel or GPU drivers` message when running `juju status` does not clear even after rebooting and clearing the notification. <!--AC-1622-->

> * `Honkai:Star Rail` game had alpha blending issues and displayed a pink rectangle during game play. <!--AC-1628-->

Reviewer: ```suggestion
* `Honkai: Star Rail` game had alpha blending issues and displayed a pink rectangle during game play. <!--AC-1628-->
```

* `Invalid argument` error occurs when configuring kernel parameters during the LXD charm upgrade. <!--AC-1645-->

> * Accessibility service crashes on Android 13 for Anbox Cloud. <!--AC-1657-->

Reviewer: This one has a public bug we should mention: https://bugs.launchpad.net/anbox-cloud/+bug/2020801

* Cannot set a node as unschedulable using the `unschedulable` parameter when updating a node via PATH method to endpoint(1.0/nodes). <!--AC-1689-->

> * Anbox Application Registry (AAR) loses metadata on upgrade. <!--AC-1691-->

Reviewer: Lets reference https://bugs.launchpad.net/anbox-cloud/+bug/2024426 even if it's not public. For a particular customer it is.

> * The LXD storage pool size is not evaluated during the initialisation of Anbox Cloud Appliance. <!--AC-1698-->

Reviewer: This is https://bugs.launchpad.net/anbox-cloud/+bug/2024559

* Applications requiring camera and microphone do not work in Anbox Cloud version 1.18.0 or later because of a change to the wire protocol between the WebRTC client and server. <!--AC-1710-->
* In the [Anbox Stream Gateway Charm](https://charmhub.io/anbox-stream-gateway), TLS pinning is not enabled even after the `use_tls_pinning` configuration is set to `true`. <!--AC-1683-->

> * The Grafana dashboard displays incorrect metrics for streaming stack. <!--AC-1685-->

Reviewer: This is https://bugs.launchpad.net/anbox-cloud/+bug/2024144

> 

Reviewer 2: As discussed last Friday, we need to call out the bug https://warthogs.atlassian.net/browse/AC-1733 in the 1.18.2 release
and regarding the [LP link](https://bugs.launchpad.net/anbox-cloud/+bug/2025058)

Author: Done

Please see [component versions](https://anbox-cloud.io/docs/component-versions) for a list of updated components.

## New features & improvements

> * Android security updates for July 2023. See [Android Security Bulletin - July 2023](tba) for more information.

Reviewer 3: ```suggestion
* Android security updates for July 2023. See [Android Security Bulletin - July 2023](https://source.android.com/docs/security/bulletin/2023-07-01) for more information.
```

> * The following security patches that were unavailable when Anbox Cloud version 1.18.1 released are included now:
>     *  CVE-2023-21115 A-258834033
>     *  CVE-2023-21121 A-205460459

Reviewer 3: ```suggestion
```
They were already present, we just weren't sure if they were, this does not need to be called in the release notes.

* The Android WebView has been updated to [tba](tba).

## Bug fixes
* When upgrading Anbox Cloud, the `Reboot required to activate new kernel or GPU drivers` message when running `juju status` does not clear even after rebooting and clearing the notification. <!--AC-1622-->
* `Honkai:Star Rail` game had alpha blending issues and displayed a pink rectangle during game play. <!--AC-1628-->
* `Invalid argument` error occurs when configuring kernel parameters during the LXD charm upgrade. <!--AC-1645-->

> * Accessibility service does not work on Android 13 for Anbox Cloud. <!--AC-1657-->

Reviewer 3: ```suggestion
* Accessibility service crashes on Android 13 for Anbox Cloud. <!--AC-1657-->
```
It was working, but it was crashing in some cases.
@adglkh Can you confirm?

* Cannot set a node as unschedulable using the `unschedulable` parameter when updating a node via PATH method to endpoint(1.0/nodes). <!--AC-1689-->
* Anbox Application Registry (AAR) loses metadata on upgrade. <!--AC-1691-->
* The LXD storage pool size is not evaluated during the initialisation of Anbox Cloud Appliance. <!--AC-1698-->

> * Applications requiring camera and microphone do not work in Anbox Cloud version 1.18.0 or later because of a change to the wire protocol between the WebRTC client and server. <!--AC-1710-->

Author: Thanks, I was wondering if that is an internal bug since the feature seems to be unused (as per Jira description). Including the bug fix now.

> * Applications requiring camera and microphone do not work in Anbox Cloud version 1.18.0 or later because of a change to the wire protocol between the WebRTC client and server. <!--AC-1710-->

Reviewer 3: https://warthogs.atlassian.net/browse/AC-1685 is also missing. This should also be customer-facing, so I think it should be included.

Author: Added it

* Applications requiring camera and microphone do not work in Anbox Cloud version 1.18.0 or later because of a change to the wire protocol between the WebRTC client and server. <!--AC-1710-->
* In the [Anbox Stream Gateway Charm](https://charmhub.io/anbox-stream-gateway), TLS pinning is not enabled even after the `tls_use_pinning` configuration is set to `true`. <!--AC-1683-->
* [LP 2024144](https://bugs.launchpad.net/anbox-cloud/+bug/2024144) The Grafana dashboard displays incorrect metrics for streaming stack. <!--AC-1685-->
* [LP 2020801](https://bugs.launchpad.net/anbox-cloud/+bug/2020801) Accessibility service crashes on Android 13 for Anbox Cloud. <!--AC-1657-->
* [LP 2024426](https://bugs.launchpad.net/anbox-cloud/+bug/2024426) Anbox Application Registry (AAR) loses metadata on upgrade. <!--AC-1691-->
* [LP 2024559](https://bugs.launchpad.net/anbox-cloud/+bug/2024559) The LXD storage pool size is not evaluated during the initialisation of Anbox Cloud Appliance. <!--AC-1698-->
* [LP 2025058](https://bugs.launchpad.net/anbox-cloud/+bug/2025058) The Anbox Cloud stream fails with an error `Anbox stream failed Error: signaling timed out`. <!--AC-1733-->

## Known issues

> * For 1.18.2 version and later, if you have a load balancer with proper TLS certificates placed in front of the Anbox stream agent and if you have set the [`location`](https://charmhub.io/anbox-stream-agent/configure#location) of the Anbox stream agent, TLS pinning will cause issues and hence you must set `tls_use_pinning=false`.

Reviewer: This is not an issue but works now as expected. Therefor we should put this under a  "Important notes" section instead.

```suggestion
* For 1.18.2 version and later, if you have a load balancer with proper TLS certificates placed in front of the Anbox stream agent and if you have set the [`location`](https://charmhub.io/anbox-stream-agent/configure#location) of the Anbox stream agent, TLS pinning MUST be disabled with `tls_use_pinning=false`. Otherwise Anbox containers will fail to communicate with the stream agent.
```

Reviewer 2: Can we give a complete command which instructs people how to disable the TLS pinning? E.g.
`juju config anbox-stream-agent tls_use_pinning=false`

Author: Done
